sentry_sdk/utils.py

```diff
@@ -1657,25 +1689,49 @@ def patch_my_function():
             return my_function()
     ```
     """
+    if TYPE_CHECKING:
+        # Type hint to ensure the default funciton has the write typing. The overloads
+        # ensure the default _no_op function is only used when R is None.
+        original_function = cast(Callable[P, R], original_function)
 
     def patcher(sentry_patched_function):
         # type: (Callable[P, R]) -> Callable[P, R]
-        @wraps(original_function)
         def runner(*args: "P.args", **kwargs: "P.kwargs"):
             # type: (...) -> R
             if sentry_sdk.get_client().get_integration(integration) is None:
                 return original_function(*args, **kwargs)
 
             return sentry_patched_function(*args, **kwargs)
 
-        return runner
+        if original_function is _no_op:
+            return wraps(sentry_patched_function)(runner)
+
+        return wraps(original_function)(runner)
 
     return patcher
 
 
+if TYPE_CHECKING:
+
+    @overload
+    def ensure_integration_enabled_async(
```

Comment:
I am confused about why `mypy` is complaining about this overload, but not the ones for `ensure_integration_enabled`. Do you have any idea? Maybe I am missing something.

Reply:
Sorry, no idea. Agree that it looks exactly like the sync case. Maybe we can just ignore the error?